### PR TITLE
feat: create rendering for inline hyperlinks

### PR DESCRIPTION
**Pull request by cezarsmpio**
This PR only renders inline links coming from an initial content.

image

image

image

image

image

image

image

### CODE REVIEW

File: packages/rich-text/src/helpers/formatDateAndTime.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// eslint-disable-next-line no-restricted-imports
```
Comment (cezarsmpio) [inline packages/rich-text/src/helpers/formatDateAndTime.ts:1]: Carried from the current editor.

File: packages/rich-text/src/helpers/truncate.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export function truncate(str, length) {
```
Comment (cezarsmpio) [inline packages/rich-text/src/helpers/truncate.ts:1]: Carried from the current editor.

File: packages/rich-text/src/plugins/Hyperlink/EntityStatusTag.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
```
Comment (cezarsmpio) [inline packages/rich-text/src/plugins/Hyperlink/EntityStatusTag.tsx:1]: Carried from the current editor.

File: packages/rich-text/src/plugins/Hyperlink/ScheduleTooltipContent.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
```
Comment (cezarsmpio) [inline packages/rich-text/src/plugins/Hyperlink/ScheduleTooltipContent.tsx:1]: Carried from the current editor.

File: packages/rich-text/src/RichTextEditor.tsx [hunk 5]
```diff
@@ -73,7 +81,155 @@ const options = {
 
 const ConnectedRichTextEditor = (props: ConnectedProps) => {
   const document = toSlatejsDocument({
-    document: props.value || Contentful.EMPTY_DOCUMENT,
+    document:
```
Comment (cezarsmpio) [inline packages/rich-text/src/RichTextEditor.tsx:84]: I will remove this later on. Just for testing purposes.

File: packages/rich-text/src/plugins/Hyperlink/EntryAssetTooltip.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            getEntityTitle: (args) => entityHelpers.getEntryTitle({ ...args, entry: args.entity }),
+          },
+        };
+
+        let contentType;
+        const entity: Entry | Asset = await actions[type].getEntity(id);
+
+        if (entity.sys.contentType) {
+          const contentTypeId = entity.sys.contentType.sys.id;
+          const contentTypes = await sdk.space.getContentTypes();
```
Comment (cezarsmpio) [inline packages/rich-text/src/plugins/Hyperlink/EntryAssetTooltip.tsx:66]: The original code uses `sdk.space.getContentType(id)` but I was getting an error from the sdk saying `getContentType` doesn't exist.

**Review by taye — APPROVED** [review]

**Review by m10l — APPROVED** [review]

File: packages/rich-text/src/plugins/Hyperlink/EntryAssetTooltip.tsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        });
+
+        const jobs = await sdk.space.getEntityScheduledActions(type, id);
+        const entityStatus = entityHelpers.getEntryStatus(entity.sys);
+
+        setEntityTitle(entityTitle);
+        setEntityStatus(entityStatus);
+        setJobs(jobs);
+      } catch (err) {
+        console.log(err);
```
Comment (djagya) [inline packages/rich-text/src/plugins/Hyperlink/EntryAssetTooltip.tsx:81]: Do we want to keep the log?

File: packages/rich-text/src/plugins/Hyperlink/ScheduleTooltipContent.tsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+};
+
+export function ScheduleTooltipContent({
+  job,
+  jobsCount,
+}: {
+  job: ScheduledAction;
+  jobsCount: number;
+}) {
+  let colorPalette = '';
```
Comment (djagya) [inline packages/rich-text/src/plugins/Hyperlink/ScheduleTooltipContent.tsx:39]: nit: Would look better as a separate function like `const colorPalette = getColor(job.action.toLowerCase())`

**Review by djagya — APPROVED** [review]
Looks good!

File: packages/rich-text/src/plugins/Hyperlink/EntryAssetTooltip.tsx (hunk 7 above)
Comment (cezarsmpio) [reply packages/rich-text/src/plugins/Hyperlink/EntryAssetTooltip.tsx:81]: We are logging it currently, so I'm not sure if we can remove it or somebody just forgot it there.

File: packages/rich-text/src/plugins/Hyperlink/ScheduleTooltipContent.tsx (hunk 8 above)
Comment (cezarsmpio) [reply packages/rich-text/src/plugins/Hyperlink/ScheduleTooltipContent.tsx:39]: I can improve it on the next PR then, this PR is green 👀

**Pull request merged**